src/urllib3/_collections.py

```diff
@@ -1,17 +1,44 @@
 import sys
 from collections import OrderedDict
-from collections.abc import Mapping, MutableMapping
-
-try:
+from typing import (
+    TYPE_CHECKING,
+    Callable,
+    Generic,
+    Iterable,
+    Iterator,
+    List,
+    Mapping,
+    MutableMapping,
+    NoReturn,
+    Optional,
+    Set,
+    Tuple,
+    TypeVar,
+    Union,
+    cast,
+    overload,
+)
+
+if TYPE_CHECKING:
```

Comment:
We assume that we can import `RLock` if we're in type-checking mode, since `threading` is apt to not exist only in environments where we're unlikely to be doing typechecking.